src/components/dataset/data/sidebar/filter/CategoryWrapper.tsx

```diff
@@ -0,0 +1,89 @@
+import React, { Dispatch, useEffect } from 'react';
+import _ from 'lodash';
+import { connect } from 'react-redux';
+import { Action } from 'redux';
+import { CHECKBOX_THRESHOLD, TableColumnType } from 'reducers/query';
+import FreetextFilter from './FreetextFilter';
+import CategoryFilterGroup from './CategoryFilterGroup';
+import { getColumnStats } from '../../../../../actions';
+import { ColumnDataTypeCategory, ResourceType } from '../../../../../constants';
+
+type CategoryWrapperProps = {
+  column: TableColumnType;
+  dataset: any;
+  dispatch: Dispatch<Action>;
+  filterMap: any;
+  filterStatement: string;
+  handleChange: () => void;
+  handleFilters: () => void;
+  tableName: string;
+  toggleExclude: () => void;
+};
+
+function CategoryWrapper({
+  column,
+  dataset,
+  dispatch,
+  filterMap,
+  handleChange,
+  handleFilters,
+  tableName,
+  filterStatement,
+  toggleExclude,
+}: CategoryWrapperProps) {
+  const refreshColumnStats =
+    column.values === undefined ||
+    (column?.originalValues && column.originalValues.length <= CHECKBOX_THRESHOLD);
+  useEffect(() => {
+    // only update for checkbox text columns on filtering
+    if (refreshColumnStats) {
+      dispatch(
+        getColumnStats(
+          ResourceType.DATASET,
+          dataset.id,
+          tableName,
+          column.name,
+          ColumnDataTypeCategory.TEXT,
+        ),
+      );
+    }
+  }, [dispatch, dataset.id, tableName, column.name, filterStatement, refreshColumnStats]);
```

Comment:
Note: We originally talked about only fetching the column stats after a user clicks on a particular field in the filter panel in order to avoid extra trips to BQ/Synapse.  I decided to leave this functionality as is for now since it would be a change in the user experience(& a little complicated to get exactly right - for example, what happens when you filter? Do you then update all fields? Or again only update on field click?), but am glad to revisit this change. 

@nmalfroy  - I'd be curious for your thoughts on this after clicking around in the updated UI.

Reply:
Ah, you called it. With Azure, the get stats calls are pretty slow and when you have a bunch of columns, the filter becomes unusable. In the example dataset from our integration tests that has 50+ columns, it takes 2 **minutes** to retrieve all of the stats. So, I think we need to make this change before we enable it for Azure.

Reply:
And, actually, it's not just Azure. It totally stalls the UI for the "transcript_consequence" table in GCP. https://jade-sh.datarepo-dev.broadinstitute.org/datasets/5ed80a11-9c2e-402c-a583-fe4c85ead4b4